### ISSUE TITLE
Soldiers will be called into buildings even if the border is divided by the sea

### ISSUE THREAD

**Issue by kyromaster** [Return-To-The-Roots/s25client#206]
Imagine the following: You have a settlement, then comes the sea and then the settlement of another player. I had the situation that my soldiers got called into the military buildings beneath the sea. They were full of soldiers because the border to the enemy was close, even if the sea was between him and me.
IMHO this shouldn't be like this. If there's (a rather large) sea between two players settlements then there should be no soldiers called to the border.

**Comment by Spikeone** [comment]
Yeah, annoying bug, also since endless maps were added because you often start in each corner on original maps.
Still for upgrading this is usefull, also mappers should know that fact and think about their map design.
Do you have a screenshot of your actual game situation? I'm quite interested in the map.

**Comment by Flow86** [comment]
yes there are many maps with problems like that (i.e Lost Worlds in the top left corner)

**Comment by Spikeone** [comment]
Would need some testing - maybe that's original. But a switch (as for endless/non endless maps) could be great. Sadly a map can be endless and still include that "bug". 
I'll test rage tomorrow

**Comment by kyromaster** [comment]
It's the "Europe" map from Settlers2 Gold. Attached is a screenshot
![bildschirmfoto vom 2015-08-15 14 11 46](https://cloud.githubusercontent.com/assets/13797213/9288935/6534631a-435f-11e5-87be-8b8a28d76f7f.png)

**Comment by Spikeone** [comment]
Working as intended:

![dosbox 2015-08-17 09-58-58-41](https://cloud.githubusercontent.com/assets/1276342/9300497/e323445e-44c6-11e5-81ec-f044085ba804.png)

**Comment by kyromaster** [comment]
So isn't this a bug but intended?
